### PR TITLE
gh-74696: Do not change the current working directory in shutil.make_archive() if possible

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
It is no longer changed when create a zip or tar archive.

It is still changed for custom archivers registered with shutil.register_archive_format()
if root_dir is not None.

#74696

### CODE REVIEW

File: Lib/shutil.py [hunk 1]
```diff
@@ -897,7 +897,7 @@ def _get_uid(name):
     return None
 
 def _make_tarball(base_name, base_dir, compress="gzip", verbose=0, dry_run=0,
-                  owner=None, group=None, logger=None):
+                  owner=None, group=None, logger=None, root_dir=None):
     """Create a (possibly compressed) tar file from all the files under
     'base_dir'.
```
Comment (merwok) [inline Lib/shutil.py:902]: Can you document the new parameter?

File: Lib/shutil.py [hunk 2]
```diff
@@ -985,42 +991,55 @@ def _make_zipfile(base_name, base_dir, verbose=0, dry_run=0, logger=None):
...
+                        arcname = os.path.join(arcdirpath, name)
+                        zf.write(path, arcname)
                         if logger is not None:
                             logger.info("adding '%s'", path)
 
+    if root_dir is not None:
+        zip_filename = os.path.abspath(zip_filename)
     return zip_filename
 
 _ARCHIVE_FORMATS = {
```
Comment (merwok) [inline Lib/shutil.py:1031]: There could be a short comment to explain what is in the format tuples here.

File: Lib/shutil.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/shutil.py:902]: It is an internal function. Most of parameters are the same as in `make_archive` where they are documented. There is no separate documentation for this function, and I think it is not needed.

File: Lib/shutil.py [hunk 3]
```diff
@@ -1217,6 +1243,11 @@ def _unpack_tarfile(filename, extract_dir):
     finally:
         tarobj.close()
 
+# Maps the name of the archive format to a tuple containing:
```
Comment (merwok) [inline Lib/shutil.py]:
```suggestion
# Maps the name of the unpack format to a tuple containing:
```

**Review by merwok — APPROVED** [review]
Looks pretty good!

New tests may be fiddly to add, but is coverage from existing tests still good?

**Comment by merwok** [comment]
Maybe change the comment added in 64f54b7ccd49

**Comment by serhiy-storchaka** [comment]
Thank you @ambv.

**Pull request merged**